src/gd_avif.c

```diff
@@ -165,6 +165,9 @@ static avifBool isAvifError(avifResult result, const char *msg) {
 
 	We ignore readFlags, just as the avifIO*ReaderRead() functions do.
 
+	In libavif 0.8.2, it's normal to get 0 bytes on the last read.
+	Thus we don't complain if charsRead < size.
```

Comment:
We don't need to add this comment because this is the specified behavior. (Also, this comment is overly specific.)

Here is the specification from avif.h:

```
// This function should return a block of memory that *must* remain valid until another read call to
// this avifIO struct is made (reusing a read buffer is acceptable/expected).
//
// * If offset exceeds the size of the content (past EOF), return AVIF_RESULT_IO_ERROR.
// * If offset is *exactly* at EOF, provide a 0-byte buffer and return AVIF_RESULT_OK.
// * If (offset+size) exceeds the contents' size, it must truncate the range to provide all
//   bytes from the offset to EOF.
// * If the range is unavailable yet (due to network conditions or any other reason),
//   return AVIF_RESULT_WAITING_ON_IO.
// * Otherwise, provide the range and return AVIF_RESULT_OK.
typedef avifResult (*avifIOReadFunc)(struct avifIO * io, uint32_t readFlags, uint64_t offset, size_t size, avifROData * out);
```
Note the third and the fifth bullet points. The truncation of the requested range mentioned in the third bullet point is done by the ctx->getBuf() call.

Reply:
I see. You can quote the specification of avifIOReadFunc as comments in this function.

For example, put the third bullet point as a comment before the ctx->getBuf() call, and put the fourth and fifth bullet points as a comment before the final return statement.